[SIEMINT-79] DDSaaS: LastPass: Crawler Integration v1.0.0 (#18225)
DataDog/integrations-core: lastpass/README.md

## Overview

[LastPass][1] is a password management solution that securely stores and manages passwords and other sensitive
information. LastPass provides users with the ability to generate passwords, synchronize passwords across multiple
devices, and add an extra layer of security through multi-factor authentication.

Integrate with Datadog to gain insights into reporting event logs through the LastPass Enterprise API. The data will be
normalized and enriched before ingestion. Pre-built dashboard visualizations provide immediate insights into LastPass
reporting events, while out-of-the-box detection rules enhance detection and response capabilities.

## Setup

### Configuration

#### Get config parameters of LastPass

##### Account Number

1. Use your email address and master password to log in and access the new [Admin Console](https://admin.lastpass.com/).
2. On the Dashboard tab, click the Profile email located in the top right corner to find the Account Number, which is
   preceded by the words "Account number".
3. Alternatively, you can find the Account Number, preceded by the words "Account number," by navigating to Advanced >
   Enterprise API.

> ##### Provisioning Hash

Author: @maycmlee Thanks for your feedback here.
As this word **Provisioning Hash** is a title for user input for installing a integration, we have kept **Hash** instead of **hash**. 
Please let us know your thoughts.

1. Access the new [Admin Console](https://admin.lastpass.com) by logging in with your email address and master password.
2. Navigate to Advanced > Enterprise API.
3. From there, you can create or reset a provisioning hash if it is forgotten.

> ##### Timezone

Reviewer: ```suggestion
##### Time zone
```

> 1. A dropdown with a list of timezones is provided. The timezone values are aligned with LastPass timezone values.

Reviewer: ```suggestion
1. The options in the **Time Zone** dropdown menu are based on LastPass' time zone values.
```

> 2. The user must select the timezone that is configured in their LastPass account.

Reviewer: ```suggestion
2. You must select the time zone that is configured in your LastPass account.
```

> 3. To verify the LastPass account timezone, perform the following steps:

Reviewer: ```suggestion
3. To verify your LastPass account's time zone, do the following:
```

    - Log in to your LastPass Business account.
    - Access the [Vault page](https://lastpass.com/vault/).

>     - Navigate to Account Settings.
>     - Under the Account Information section, you can find the selected timezone.

Reviewer: ```suggestion
    - Navigate to **Account Settings**.
    - Find the selected time zone under the **Account Information** section.
```

> #### LastPass DataDog Integration Configuration

Reviewer: ```suggestion
#### Configure the LastPass and Datadog integration
```

Configure the Datadog endpoint to forward LastPass logs to Datadog.

> 1. Navigate to `LastPass`.

Reviewer: Is this is in Datadog or LastPass?

Author: It means: **Navigate to LastPass integration on DataDog platform.**
We will refactor the statement.

2. Add your LastPass credentials.

| LastPass Parameters | Description                                                      |
|---------------------|------------------------------------------------------------------|

> | Account Number      | Account Number of your registered LastPass account.              |
> | Provisioning Hash   | Provisioning Hash secret of your registered account on LastPass. |
> | Timezone            | Timezone of your registered account on LastPass                  |

Reviewer: ```suggestion
| Account number      | The account number of your registered LastPass account.              |
| Provisioning hash   | The provisioning hash secret of your registered account on LastPass. |
| Time zone            | The time zone of your registered account on LastPass.                  |
```

## Data Collected

### Logs

> The LastPass integration collects and forwards LastPass Reporting Event logs to Datadog.

Reviewer: ```suggestion
The LastPass integration collects and forwards LastPass reporting event logs to Datadog.
```
